### ISSUE TITLE
Router's navigate() throws exception for component has no route config (in Alpha 38)

### ISSUE THREAD

**Issue by johnpapa** [angular/angular#4490]
Just tried alpha 38 … found a few minor things, but this one is confusing on the router. We can now use `navigate` or `navigateByUrl`/ The latter accepts a string (the old style) and works. The former accepts a tuple, and tosses an error at me.

```
  // this works
  this._router.navigateByUrl(`detail/${this.selectedHero.id}`);

  // this does not work
  this._router.navigate([`./${Routes.detail.as}`, {id: this.selectedHero.id}]);
```

The error is `ORIGINAL EXCEPTION: Component "HeroesComponent" has no route config.`

See example here

https://github.com/johnpapa/angular2-tour-of-heroes/blob/master/src/app/heroes.component.ts#L38-L39

cc/ @btford:

**Comment by johnpapa** [comment]
if i change the tuple to `this._router.navigate([`/${Routes.detail.as}`, {id: this.selectedHero.id}]);` - it seems to work

notice i am not prefixing with a dot this time.

Should the dot matter?

if so, the error message should be clearer.

**Comment by btford** [comment]
Yes– `/` means relative to the root of the application, and `./` means a child of this component.

So this is expected behavior, but the exception here could be a bit better.

**Comment by johnpapa** [comment]
Thanks for the explanation. Yes, I think the error message and the docs should be more clear on this. Agreed.

I have some feedback from a recent hack night on how the router might be easier for folks to use. I'll gather them and send them in a more coherent form later.

**Comment by btford** [comment]
Sounds good, much appreciated!

**Comment by btford** [comment]
Update: routes without a leading slash have landed #4623
